rt/rt.mjs

```diff
@@ -183,6 +185,10 @@ function rt(engine) {
         '%flush-log': () => {
             console.info(engine.log);
             engine.log = "";
+        },
+        '%open-as-stdin': (filename) => {
+            filename = schemeToJS(filename);
+            engine.setCurrentInputPort(fs.readFileSync(filename));
```

Comment:
I think instead when instantiating the engine, the instantiator should provide some kind of filesystem interface.  This would allow the instantiator to e.g. not grant any filesystem access, or to use some API exported by the raw shells (`d8`, `jsc`, etc), or to use Web APIs.  WDYT?  Otherwise I think this change is a problem, as it adds Node-specific API here.

Reply:
Yeah, I think this is a good idea. We should keep the engine independent of whatever platform it's running on.

I'll rework this with a simple file system API. I'm thinking something like `fs.open(filename)`, which returns a `File` object with methods like `readAll`. That'll be enough to make progress here, but we'll want to expand/rework it as we improve file system support.

Reply:
I just pushed another change that adds a more abstract filesystem interface.